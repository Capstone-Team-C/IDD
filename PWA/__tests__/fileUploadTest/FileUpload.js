import {shallowMount} from '@vue/test-utils'
import FileUpload from '../../src/components/Timesheet/FileUploader.vue'
import { mount } from '@vue/test-utils'
import sinon from 'sinon'

describe('FileUpload', () => {

	//Checks if the upload status is not active since no files to upload
	it('upload status should be zero since no files', () => {
		const wrapper = mount(FileUpload, {
			propsData: {
				files: []
			}
		})
		expect(wrapper.vm.files.length).toBe(0)
})

	//Checks if there is a file, and the "Complete timesheet" button was clicked, the loader opens.
	/*
	it('checks loader is called when complete timesheet button is clicked' () => {
		let mockFile = Object.create(null)
		wrapper = shallowMount (FileUpload, {
			propsData: {
				files: [mockFile]
			}
		})
*/

})
