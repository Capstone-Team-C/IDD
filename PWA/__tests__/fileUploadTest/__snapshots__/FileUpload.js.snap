// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload should't render since files are empty 1`] = `
<div
  class="example-drag"
  files=""
>
  <div
    class="upload"
  >
    <ul>
      <td
        colspan="7"
      >
        <div
          class="text-center p-5"
        >
          <h4>
            Drop files anywhere to upload
            <br />
            or
          </h4>
           
          <label
            class="btn btn-lg btn-primary"
            for="file"
          >
            Select Files
          </label>
        </div>
      </td>
    </ul>
     
    <div
      class="drop-active"
      style="display: none;"
    >
      <h3>
        Drop files to upload
      </h3>
    </div>
     
    <div
      class="example-btn"
    >
      <file-upload-stub
        accept="image/*, application/pdf"
        chunk="[object Object]"
        class="btn btn-primary"
        data="[object Object]"
        drop="true"
        dropdirectory="true"
        extensions=""
        headers="[object Object]"
        maximum="2"
        multiple="true"
        name="file"
        postaction="https://en91d9jkz3gou.x.pipedream.net"
        size="10485760"
        thread="1"
        timeout="0"
        value=""
      >
        <i
          class="fa fa-plus"
        />
        
          Select files
        
      </file-upload-stub>
       
      <button
        class="btn btn-success"
        type="button"
      >
        <i
          aria-hidden="true"
          class="fa fa-arrow-up"
          id="startUpload"
        />
        
          Start Upload
        
      </button>
    </div>
  </div>
</div>
`;

exports[`FileUpload shouldn't render files since bad file 1`] = `
<div
  class="example-drag"
  files="[object Object]"
>
  <div
    class="upload"
  >
    <ul>
      <td
        colspan="7"
      >
        <div
          class="text-center p-5"
        >
          <h4>
            Drop files anywhere to upload
            <br />
            or
          </h4>
           
          <label
            class="btn btn-lg btn-primary"
            for="file"
          >
            Select Files
          </label>
        </div>
      </td>
    </ul>
     
    <div
      class="drop-active"
      style="display: none;"
    >
      <h3>
        Drop files to upload
      </h3>
    </div>
     
    <div
      class="example-btn"
    >
      <file-upload-stub
        accept="image/*, application/pdf"
        chunk="[object Object]"
        class="btn btn-primary"
        data="[object Object]"
        drop="true"
        dropdirectory="true"
        extensions=""
        headers="[object Object]"
        maximum="2"
        multiple="true"
        name="file"
        postaction="https://en91d9jkz3gou.x.pipedream.net"
        size="10485760"
        thread="1"
        timeout="0"
        value=""
      >
        <i
          class="fa fa-plus"
        />
        
          Select files
        
      </file-upload-stub>
       
      <button
        class="btn btn-success"
        type="button"
      >
        <i
          aria-hidden="true"
          class="fa fa-arrow-up"
          id="startUpload"
        />
        
          Start Upload
        
      </button>
    </div>
  </div>
</div>
`;
